src/app/speechtotext/speechtotext.component.ts

```diff
@@ -86,7 +86,8 @@ export class SpeechtotextComponent implements OnInit {
 
       if (t === 5) {
         this.subscription.unsubscribe();
-        this.store.dispatch(new speechactions.SearchAction(false));
+        this.message = "Please check your microphone and audio levels.";
+        // this.store.dispatch(new speechactions.SearchAction(false));
```

Comment:
great work @harshit98 uncomment this line. but dispatch the action when t==7.
this action automatically closes the speech UI and returns to the index page.